### DISCUSSION TITLE
double click does not work sometimes

### DISCUSSION THREAD

**Discussion by ZhouMM92** [sourcelair/xterm.js#4616] [Q&A]
I want to use double-click event in xterm.js for automatically copy and paste, but double-click event does not work when there is words in the area where the mouse click. But double-click event works when there is empty in the area where the mouse click. The codes are as follows:
```
<!-- myTerm.vue -->
<template>
<div :id="term_id"   @dblclick="copyAndPaste()" ></div>
</template>

<script>
export default {
  mounted(){
      this.term = new Terminal({
      disableStdin: false,
      rightClickSelectsWord: true
})
     this.term.open(document.getElementById(this.term_id))
},
methods:{
  copyAndPaste(){
    console.log("copyAndPaste")
    console.log(this.term.getSelection())
}}}
</script>
```
As you can see, when I double-click "22:34:06", there are nothing in console:
![image](https://github.com/xtermjs/xterm.js/assets/101792358/fea6896c-941a-4d2e-9656-8bd22e1bb3e4)

But when I double-click the empty area, there are the outputs in console:
![image](https://github.com/xtermjs/xterm.js/assets/101792358/fc1d97d1-ded1-494b-a4c0-7fe41b1837ee)

It's still unclear to me why/when this would happen. So I will be very appreciated if you can help me find the reason.
## Details
- Browser and browser version: chorme
- OS version: windows 10
- xterm.js version: 5.1.0

**Comment by ZhouMM92** [comment]
@Tyriar

**Comment by ZhouMM92** [comment]
@jerch

**Comment by Tyriar** [comment]
You cannot just hook up regular mouse events as they're handled internally by xterm.js and not necessarily bubbled up to its container.

I'm actually not sure what the best way to handle double click is, but using the `onSelectionChange` will let you know when anything is selected.
